velox/experimental/cudf/connectors/parquet/ParquetDataSource.cpp

```diff
@@ -250,8 +251,27 @@ void ParquetDataSource::totalScanTimeCalculator(void* userData) {
 }
 
 void ParquetDataSource::addSplit(std::shared_ptr<ConnectorSplit> split) {
-  // Dynamic cast split to `ParquetConnectorSplit`
-  split_ = std::dynamic_pointer_cast<ParquetConnectorSplit>(split);
+  split_ = [&]() {
+    // Dynamic cast split to `ParquetConnectorSplit`
+    if (std::dynamic_pointer_cast<ParquetConnectorSplit>(split)) {
+      return std::dynamic_pointer_cast<ParquetConnectorSplit>(split);
+      // Convert `HiveConnectorSplit` to `ParquetConnectorSplit`
+    } else if (std::dynamic_pointer_cast<hive::HiveConnectorSplit>(split)) {
+      const auto hiveSplit =
+          std::dynamic_pointer_cast<hive::HiveConnectorSplit>(split);
+      VELOX_CHECK_EQ(
+          hiveSplit->fileFormat,
+          dwio::common::FileFormat::PARQUET,
+          "Unsupported file format for conversion from HiveConnectorSplit to ParquetConnectorSplit");
```

Comment:
Should this say cudf ParquetConnectorSplit ?

Reply:
Thanks for the suggestion. Updated